cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,75 @@
 /// <reference types='cypress' />
-
+import { generateUser } from '../support/generate';
 describe('Student Registration page', () => {
   before(() => {
-
+    cy.visit('https://demoqa.com/automation-practice-form');
   });
 
-  it('', () => {
+  const { email, firstName, lastName, address } = generateUser();
+  const number = 1234567891;
+
+  it('should enter vaild data to register', () => {
+    cy.get('#firstName').type(firstName);
+
+    cy.get('#lastName').type(lastName);
+
+    cy.get('#userEmail').type(email);
+
+    cy
+      .get('#genterWrapper > .col-md-9 > :nth-child(1) > .custom-control-label')
+      .click();
+
+    cy.get('#userNumber').type(number);
+
+    cy.get('.subjects-auto-complete__value-container').type('Eng');
+
+    cy.get('#react-select-2-option-0').click();
+
+    cy.get('#state').click();
+
+    cy.get('#react-select-3-option-0').click();
+
+    cy.get('#city').click();
+
+    cy.get('#react-select-4-option-0').click();
+
+    cy.get('#hobbiesWrapper > .col-md-9 > :nth-child(1)').click();
+
+    cy.get('#hobbiesWrapper > .col-md-9 > :nth-child(2)').click();
+
+    cy.get('#hobbiesWrapper > .col-md-9 > :nth-child(3)').click();
+
+    cy.get('#currentAddress').type(address);
+
+    cy.get('#submit').click();
```

Comment:
After clicking the submit button, it would be a good practice to check if the form submission was successful. For example, you could check if a success message is displayed or if the user is redirected to a different page.